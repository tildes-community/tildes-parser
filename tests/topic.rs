use std::fs::read_to_string;

use {insta::assert_debug_snapshot, tildes_parser::Topic};

#[test]
fn test_topic_parsing() {
  let samples = ["link", "text", "deleted", "scheduled"];
  for sample in samples {
    let html =
      read_to_string(format!("tests/samples/topic_{sample}.html")).unwrap();
    let topic = &html.parse::<Topic>().unwrap();
    assert_debug_snapshot!(sample, topic);
  }
}
