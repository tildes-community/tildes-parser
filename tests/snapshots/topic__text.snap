---
source: tests/topic.rs
expression: topic
---
Topic {
    author: Name(
        "User",
    ),
    comment_total: 15,
    content: Text(
        "<p>Topic Text</p>",
    ),
    id: "123",
    is_locked: true,
    is_official: false,
    tags: [
        "one",
        "two",
        "three",
    ],
    vote_count: 5,
}
